bio/megahit/wrapper.py

```diff
@@ -0,0 +1,107 @@
+"""Snakemake wrapper for megahit."""
+
+__author__ = "Jie Zhu @alienzj"
+__copyright__ = "Copyright 2025, Jie Zhu"
+__email__ = "alienchuj@gmail.com"
+__license__ = "MIT"
+
+import os, shutil
+from snakemake.shell import shell
+
+
+# infer output directory
+
+if hasattr(snakemake.output, "dir"):
+    output_dir = snakemake.output.dir
+
+else:
+    # get output_dir file from output
+    if hasattr(snakemake.output, "contigs"):
+        output_file = snakemake.output.contigs
+    else:
+        output_file = snakemake.output[0]
+
+    output_dir = os.path.split(output_file)[0]
```

Comment:
I think it is best that all output files have to be specified, so that `snakemake` can check their existence.

```suggestion
output_dir = os.path.split(snakemake.output[0])[0]
```

Reply:
Actually, if you use `tempfile`, there is no need to define `output_dir`.

Reply:
Based on the `megahit` output in the assembly directory, I believe the `contigs.fasta` file alone would be sufficient (or: specifying just the contigs output directory would be enough too).

```
λ ll assembly
  inode Permissions  Size User    Group Date Modified Date Accessed Name
3666787 .rw-r-----    262 alienzj users 20 May 11:09  20 May 11:08  checkpoints.txt
3666976 .rw-r-----  902Ki alienzj users 20 May 11:09  20 May 11:09  contigs.fasta
3666977 .rw-r-----      0 alienzj users 20 May 11:09  20 May 11:09  done
3666783 drwxr-x---      - alienzj users 20 May 11:09  20 May 11:08  intermediate_contigs
3666784 .rw-r-----  138Ki alienzj users 20 May 11:09  20 May 11:08  log
3666785 .rw-r-----  1.1Ki alienzj users 20 May 11:08  20 May 11:08  options.json
```

Reply:
But maybe in the future other users also need other files, like `options.json` or `log`.